In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

sns.set_style("whitegrid")

from pathlib import Path
import sys

notebook_dir = Path().resolve()
common_utils_path = notebook_dir.parent / "common_utils"
sys.path.append(str(common_utils_path))

from EDA import dataset_overview,categorical_analysis,num_analysis

In [2]:
df = pd.read_csv('data/train.csv')
df_test = pd.read_csv('data/test.csv')
df_final = pd.read_csv('data/sample_submission.csv')

In [3]:
dataset_overview(df)

==================================== Dataset Overview ====================================

============ Data Shape ============
Rows: 90615
Columns: 10


============ Datatypes ============
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90615 entries, 0 to 90614
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              90615 non-null  int64  
 1   Sex             90615 non-null  object 
 2   Length          90615 non-null  float64
 3   Diameter        90615 non-null  float64
 4   Height          90615 non-null  float64
 5   Whole weight    90615 non-null  float64
 6   Whole weight.1  90615 non-null  float64
 7   Whole weight.2  90615 non-null  float64
 8   Shell weight    90615 non-null  float64
 9   Rings           90615 non-null  int64  
dtypes: float64(7), int64(2), object(1)
memory usage: 6.9+ MB
============ Missing Values ============


id                0
Sex               0
Length            0
Diameter          0
Height            0
Whole weight      0
Whole weight.1    0
Whole weight.2    0
Shell weight      0
Rings             0
dtype: int64



============ Duplicates Values ============
Duplicated values : 0
============ Data Preview ============
Head:


id Sex  Length  Diameter  Height  Whole weight  Whole weight.1  \
0   0   F    0.55      0.43   0.150        0.7715          0.3285   
1   1   F    0.63      0.49   0.145        1.1300          0.4580   
2   2   I    0.16      0.11   0.025        0.0210          0.0055   

   Whole weight.2  Shell weight  Rings  
0          0.1465         0.240     11  
1          0.2765         0.320     11  
2          0.0030         0.005      6

Tail:


id Sex  Length  Diameter  Height  Whole weight  Whole weight.1  \
90612  90612   I   0.435     0.330   0.095        0.3215          0.1510   
90613  90613   I   0.345     0.270   0.075        0.2000          0.0980   
90614  90614   I   0.425     0.325   0.100        0.3455          0.1525   

       Whole weight.2  Shell weight  Rings  
90612          0.0785        0.0815      6  
90613          0.0490        0.0700      6  
90614          0.0785        0.1050      8

Sample:


id Sex  Length  Diameter  Height  Whole weight  Whole weight.1  \
9690    9690   I   0.365     0.280   0.105        0.2785          0.1205   
74994  74994   I   0.560     0.440   0.125        0.8565          0.3595   
4598    4598   F   0.585     0.465   0.170        1.0525          0.5160   

       Whole weight.2  Shell weight  Rings  
9690           0.0565        0.0750      8  
74994          0.1715        0.2385      9  
4598           0.2435        0.2895      9



============ Numerical and Categorical Values ============
Numerical Datatypes: Index(['id', 'Length', 'Diameter', 'Height', 'Whole weight', 'Whole weight.1',
       'Whole weight.2', 'Shell weight', 'Rings'],
      dtype='object')
Number of numeric features: 9
Categorical Datatypes: Index(['Sex'], dtype='object')
Number of categorical features: 1


(Index(['id', 'Length', 'Diameter', 'Height', 'Whole weight', 'Whole weight.1',
        'Whole weight.2', 'Shell weight', 'Rings'],
       dtype='object'),
 Index(['Sex'], dtype='object'))

# Observations
1. There are 7 numeric (1 id column and 1 target feature) features and 1 categorical feature
2. No missing values or duplicated values
3. 90615 rows of data